paimon-core/src/main/java/org/apache/paimon/table/sink/TableCommitImpl.java

```diff
@@ -120,7 +120,9 @@ public TableCommitImpl(
     }
 
     public boolean forceCreatingSnapshot() {
-        return tagAutoCreation != null && tagAutoCreation.forceCreatingSnapshot();
+        return tagAutoCreation != null
+                && tagAutoCreation.canProceedWithNextTag()
```

Comment:
`forceCreatingSnapshot` first and then `canProceedWithNextTag`?

Reply:
done.